### PR TITLE
New Backoffice: Log viewer controller

### PR DESCRIPTION
**Pull request by elit0451**
## Details
- Created a new service for the LogViewer that gathers implementations from `ILogViewer`, `ILogViewerConfig` and `ILogLevelLoader`;
- New endpoints for the Log Viewer:
  - ~GET `/log-viewer/error/count` - gets the number of errors from the logs;~ _(removed in commit c470f6cda12333e72706be0906c71966a59a4d59 since the number can be derived from `/log-viewer/level-count`, summing Error + Fatal)_;
  - GET `/log-viewer/level` - gets all loggers' levels in a paginated list;
  - GET `/log-viewer/level-count` - gets the count for each log level from the logs;
  - GET `/log-viewer/log` - gets all logs in a paginated list;
    ~- A temp solution for representing enums as strings for solving an issue with Swagger has been added in commit c6e4ac35f7b228da879c66d2d75380a9159b7f7f;~ If you want to test with selecting the enums types, it has to be in Postman;
    - Note ❗ there is currently a polymorphism issue that we will work on in the future, mentioned in [comment](https://github.com/umbraco/Umbraco-CMS/blob/v13/new-backoffice/log-viewer-controller/src/Umbraco.Infrastructure/Services/Implement/LogViewerService.cs#L219-L220). The problem we see is in the properties of some log entries with Error log level, like:
 > Before:
```json
{
  . . .
  "items": [
    {
      . . .
      "Level": "Error",
	  . . .
      "Properties": {
        "EventId": {
          "TypeTag": null,
          "Properties": [
            {
              "Name": "Id",
              "Value": {
                "Value": 1
              }
            },
            {
              "Name": "Name",
              "Value": {
                "Value": "UnhandledException"
              }
            }
          ]
        }
      }
    }
  ]
}
```
 > Now (_The nested `Value` object is empty_):
```json
{
  . . .
  "items": [
    {
      . . .
      "level": "Error",
	  . . .
      "properties": [
        {
          "name": "EventId",
          "value": "{\"TypeTag\":null,\"Properties\":[{\"Name\":\"Id\",\"Value\":{}},{\"Name\":\"Name\",\"Value\":{}}]}"
        }
      ]
    }
  ]
}
```
- 
  - GET `/log-viewer/message-template` - gets all log message templates (_used to determine common log patterns or errors_) in a paginated list;
  - GET `/log-viewer/saved-search` - gets all saved log searches in a paginated list;
  - POST `/log-viewer/saved-search` - creates a saved log search;
  - GET `/log-viewer/saved-search/{name}` - gets a saved log search by name;
  - DELETE `/log-viewer/saved-search/{name}` - deletes a saved log search with a given name;
  - GET `/log-viewer/validate-logs-size` - gets a response on whether you are able to view logs for a specified date range (<em><code>true</code> when the requested log files are less than 1GB in size</em>)
- Cherry-pick  #13634 and #13647;
- Introducing `GetLogsAsPagedModel()`+ update for it in a _suppressions.xml_ file;

## Test
- You can use Swagger to test the endpoints - visit `localhost/umbraco/swagger`;
~- `/log-viewer/error/count` - _GET_~
  ~- Compare the number of errors for a given time period, like `startDate=2023-01-09` and `endDate=2023-01-10`, with the one shown on the _Log Viewer_ dashboard.~
- `/log-viewer/level` - _GET_
  - Execute with the default _skip_ and _take_ values or modify them as you like;
  - Compare with the response from the backoffice when going to the _Log Viewer_ - _the response from `GetLogLevels()`_.
- `/log-viewer/level-count` - _GET_
  - Compare with the response from the backoffice when going to the _Log Viewer_ - _the response from `GetLogLevelCounts()`_.
- `/log-viewer/log` - _GET_
  - Compare with the response (`items[]`) from the backoffice when going to the _Log Viewer_ > _Saved Searches (All Logs)_;
    - You can filter the logs by log level, like `logLevel=Error&logLevel=Fatal`;
      - _Note:_ in Swagger you can select several levels by holding _Ctrl_.
    - Or add a filter expression, like `Not(@Level='Verbose') and Not(@Level='Debug')` (_check the default Saved Searches for other examples_);
    - etc.
- `/log-viewer/message-template` - _GET_
  - Compare with the response from the backoffice when going to the _Log Viewer_ - _the response from `GetMessageTemplates()`_;
  - They can be seen in _Log Viewer_ > _Common Log Messages_.
- `/log-viewer/saved-search` - _GET_
  - Compare with the response from the backoffice when going to the _Log Viewer_ - _the response from `GetSavedSearches()`_;
  - They can be seen in _Log Viewer_ > _Saved Searches_.
- `/log-viewer/saved-search` - _POST_
  - Create a new saved search by providing a name and a query value, like:
```json
{
  "name": "Find all logs where the Level is NOT Information and NOT Warning",
  "query": "Not(@Level='Information') and Not(@Level='Warning')"
}
```
-
  - Verify that the response headers contain a **Location** response header, indicating the URL to get the details about this item;
  - Try creating a saved search with a name that already exists - the response should be with status 400 and title _"Duplicate log search name"_.
- `/log-viewer/saved-search/{name}` - _GET_
  - _New endpoint_: get all saved searches from GET `/log-viewer/saved-search` and find a name of a saved search;
  - Check if the response contains the right query.
- `/log-viewer/saved-search/{name}` - _DELETE_
  - Pass the name of a saved search;
  - Verify that if the item exists the response is OK, otherwise it will be Not Found.
- `/log-viewer/validate-logs-size` - _GET_
  - Compare with the response from the backoffice when going to the _Log Viewer_ - _the response from `GetCanViewLogs()`_.

### CODE REVIEW

File: src/Umbraco.Infrastructure/Logging/Viewer/SerilogLogViewerSourceBase.cs [hunk 1]
```diff
@@ -60,13 +71,80 @@ public IEnumerable<LogTemplate> GetMessageTemplates(LogTimePeriod logTimePeriod)
...
+    }
+
+    [Obsolete("Use ILogViewerService.GetPagedLogs instead. Scheduled for removal in Umbraco 15.")]
+    public PagedModel<LogMessage> GetLogsAsPagedModel(
+        LogTimePeriod logTimePeriod,
+        int skip,
+        int take,
+        Direction orderDirection = Direction.Descending,
+        string? filterExpression = null,
+        string[]? logLevels = null)
```
[the remark this answers is not in the document]
Comment (elit0451) [reply src/Umbraco.Infrastructure/Logging/Viewer/SerilogLogViewerSourceBase.cs:107]: As we agreed, we will have a refactoring task before Umbraco 13 goes out.

File: src/Umbraco.Infrastructure/Services/Implement/LogViewerService.cs [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    {
+        LogTimePeriod logTimePeriod = GetTimePeriod(startDate, endDate);
+
+        // We will need to stop the request if trying to do this on a 1GB file
+        if (CanViewLogs(logTimePeriod) == false)
+        {
+            return Attempt<LogLevelCounts>.Fail();
+        }
+
+        return Attempt<LogLevelCounts>.Succeed(_logViewer.GetLogLevelCounts(logTimePeriod));
```
[the remark this answers is not in the document]
Comment (elit0451) [reply src/Umbraco.Infrastructure/Services/Implement/LogViewerService.cs]: As we agreed, we will have a refactoring task before Umbraco 13 goes out.

File: umbraco.sln [hunk 3]
```diff
@@ -34,29 +34,6 @@ Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "Umbraco.Web.UI.Client", "ht
 		StartServerOnDebug = "false"
 	EndProjectSection
 EndProject
-Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "Umbraco.Web.UI.New.Client", "http://localhost:3961", "{17E2067E-9D0F-4891-B6BA-CD828886B968}"
```
Comment (nikolajlauridsen) [inline umbraco.sln:37]: Looks like some merge issues here, the `Umbraco.Web.UI.New.Client` project is missing from the solution

**Review by nikolajlauridsen — COMMENTED** [review]
Looks good now, and tests good now 💪 good work

Only found one thing that we should fix before merging, seems like something went wrong when merging the solution file

File: umbraco.sln (hunk 3 above)
Comment (nikolajlauridsen) [reply umbraco.sln:37]: Or is this intentional and I'm just missing something? :)

Comment (elit0451) [reply umbraco.sln:37]: Yeah, I saw that but my IDE suggested it twice to me, even though I discarded the changes. But let me have a look again 😉

**Review by nikolajlauridsen — APPROVED** [review]
Looks all good to me now 👍

**Pull request merged**